core/shared/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -141,7 +139,7 @@ object EntityEncoder {
 
   implicit def stringEncoder(implicit charset: Charset = `UTF-8`): EntityEncoder.Pure[String] = {
     val hdr = `Content-Type`(MediaType.text.plain).withCharset(charset)
-    simple(hdr)(s => Chunk.array(s.getBytes(charset.nioCharset)))
+    simple(hdr)(ByteVector.encodeString(_)(charset.nioCharset).toOption.get)
```

Comment:
So, this is a bit weird. Am I missing something, should this be a `Pure` encoder?

Reply:
I'm slightly inclined towards making it an impure encoder to get rid of this `.toOption.get`.